File: zad09.04.15/zad2.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

""" Alorytm rank dla generowania wszystkich podzbiorow dlugosci k
w porzadku leksykograficznym."""

def get_bin_vector(t):
    bin_vector = 0
    for i in t:
        bin_vector |= 1 << i - 1
    return bin_vector


def start():
    t = raw_input("Podaj liste: ").split(' ')
    t = sorted([int(x) for x in t])

    n = int(raw_input("Podaj n: "))

    result = get_bin_vector(t)
    print result if result < 1 << n else "Wrong data"

if __name__ == '__main__':
    start()

File: zad14.06/zad2.c
#include <stdio.h>
#include <iostream>

using namespace std;

int main()
{
    int rank, n;
    cout << "Podaj rank: ";
    cin >> rank;
    
    cout << "Podaj n: ";
    cin >> n;

    int b[n+1];
    for(int i = 0; i <= n; i++) {
        b[i] = 0;   
    }
    
    int i = 0;
    while (rank) {
        b[i] = rank % 2 ? 1 : 0;
        rank /=2 ;
        i++;
    }
    
    cout << "{ ";
    
    for(i = 0; i < n; i++) {
           if (((b[i] + b[i+1]) % 2) == 1) {
                cout << i + 1 << " ";   
           }
    }
    
    cout << "}" << endl;
    return 0;
}